architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst

```diff
@@ -3,14 +3,55 @@ Heap sharing
 
 The memory heap can be:
 
--  private - attached to one of the DSPs,
+-  local - used exclusively by a single DSP core,
 -  shared - higher level memory shared across all DSP cores
 
-The shared memories are co-managed by all DSP cores that have access to it.
+.. uml:: images/heaps.pu
+   :caption: Memory Heaps
 
-The data structures needed to manage shared memories are initialized by primary
-core, structure location in memory map is known at the build time and access to
-it is controlled by mutex.
+.. note:: Introduction of MMU will require a separate local application heap per
+   isolated domain.
+
+L1 Cache Coherency
+******************
+
+NOTE: This section applies to systems without L1 cache coherency.
+
+A local heap is used exclusively by a single DSP core. Therefore operations on
+the allocated memory buffers do not require explicit L1 cache operations nor
+data cache alignment.
+
+All operations performed on a local heap can be executed by the associated DSP
+core only. The *move-to-another-core* operation is not permitted for allocated
+buffers.
+
+A shared heap can be configured in two ways:
+
+1. To provide uncache aliases of buffer addresses to the clients,
+2. To provide cacheable addresses to the clients.
+
+Option #1 is preferred, since does not require explicit L1 cache operations
+when memory is accessed by a DSP core. However all operations directly access
+L2+ memory therefore it is not suitable for a low latency high performance data
+processing case.
+
+Option #2 provides better performance but requires explicit L1 cache operations,
+which are difficult to maintain and validate, as well as data cache alignment
+for both client buffers and their descriptors, which creates an overhead. This
+configuration should be avoided if possible.
```

Comment:
but it is possible to share data across core L1 cache via the coherent API.

Reply:
I have added this info to the description